### PR TITLE
Adding security + IAC suggestions for SQL DB security rec's

### PR DESCRIPTION
**Pull request by NWiddup**
Adding security + IAC suggestions, around implementing IAC to create the platform and then using IAC to create/rotate static secrets & store them in Key Vault.

### CODE REVIEW

**Comment by NWiddup** [comment]
This is WIP & will need a bit of extra detail added (e.g. reference links & GUID for rec). I've got a couple of other things I would like to add to this as well, but creating the PR to discuss these items.

#on-hold

**Comment by NWiddup** [comment]
Some of the items to consider adding are:
- MFA enforcement
- Conditional access policies to restrict auth only from approved devices/endpoints/locations/regions/countries
- [SQL DB Security handbook](https://learn.microsoft.com/en-us/azure/azure-sql/database/security-overview?view=azuresql)
- incorporate relevant [Best Practice Recommendations for Azure SQL DB](https://learn.microsoft.com/en-us/azure/azure-sql/database/security-best-practice?view=azuresql)
- [Azure SQL Security Baseline](https://learn.microsoft.com/en-us/security/benchmark/azure/baselines/azure-sql-security-baseline?toc=%2Fazure%2Fazure-sql%2Ftoc.json&view=azuresql)

**Comment by erjosito** [comment]
@igorpag could you please review?

**Comment by igorpag** [comment]
Hi @NWiddup, thanks for your contribution. Regarding the content you proposed to include, that is SQL DB Security handbook, Best Practices Recommendations for Azure SQL DB and Azure SQL Security Baseline, I have already reviewed these repos, then everything relevant should be already included. If there is anything specific I missed, maybe, please let me know and we will include. Regarding MFA enforcement, CA, and the two items you just added in the PR, I'm not sure this security checklist is the right place: we discussed already inside the broader vTeam, and this kind of generic (from a SQL point of view) items should be covered in a separate checklist. In the specific case, we should have a DevOps checklist. Let me bring into the discussion @xstof  and @jelledruyts ..... what you think guys? Thanks.

**Comment by jelledruyts** [comment]
I agree with Igor that the "Use infrastracture as code to automate the deployment of your SQL DB infrastructure" item is indeed a very general-purpose recommendation that shouldn't be repeated in every checklist (especially not a security-focused one).

The other two around creating and rotating SQL platform secrets could be a single one to me, and even then I have some reservations:
- It's not specific to SQL as it would generally also be similar for any kind of "local" authentication solution (like SAS for storage, or API keys for Search etc) so it could be part of a governance checklist that "for any service where you have control over a local credential (like SQL's SA account in this case), ensure you (re)generate and store it appropriately".
- _If_ we feel strong enough that it really needs to be called out specifically for SQL, I'd first make sure to note that it's only relevant if you choose to ignore the other checklist items that recommend using AAD auth (meaning, if you turn on _AAD only auth_, then you don't even _need_ an SA account/password that needs to be properly generated and stored in the first place).
- And even with that caveat, I wouldn't lead with saying "how" a customer should do that (which is an implementation detail); you could _suggest_ to use a DevOps pipeline, but the important part is that it's securely generated and stored - not how that's done.

So my take on this would be: _if_ we want to call out the SA password very explicitly, I would rephrase it to say something like "avoid it by using AAD auth if at all possible, but if you _must_ have it then generate and store it appropriately (for example, using a devops pipeline + key vault)".

@igorpag @NWiddup @xstof @erjosito

**Comment by xstof** [comment]
Agree with both Igor and Jelle.  In addition I wanted to note that there's an explicit goal of having items that have single yes/no answer and that are actionable.  So for example, we're trying to avoid pointing to a document with several items of guidance and rather want to include them directly in the checklist as individual items so it's much more tanglible and actionable for customers.  I believe that's what Igor did with the security handbook.

**Comment by igorpag** [comment]
Available to continue the discussion "offline".
